core/nixpkgs.bzl

```diff
@@ -884,3 +884,92 @@ def nixpkgs_flake_package(
     )
 
     _nixpkgs_flake_package(**kwargs)
+
+def _run_nix_shell_impl(ctx, **kwargs):
+    out_file = ctx.actions.declare_file("%s" % ctx.attr.output)
+
+    nix_env = ctx.file.stdenv
+
+    ctx.actions.run_shell(
+      inputs = [nix_env],
+      outputs = [out_file],
+      progress_message = "Running nix shell...",
+      arguments = [out_file.path],
+      command = """
+        source {}
+        {}
+        """.format(nix_env.path, ctx.attr.cmd, out_file.path)
```

Comment:
This is strange, you are passing three arguments to format, while there are only two placeholders.